run/john.conf

```diff
@@ -3497,6 +3497,42 @@ void restore()
 	}
 }
 
+# generate all possible wps pins
+[List.External:wpspin]
+int pin;
+void init() {
+	pin = 0;
+}
+void generate() {
+	if(pin > 9999999) {
+		word[0] = 0;
+		return;
+	}
+	int i, p;
+	i = 0;
+	while(i < 8) word[i++] = '0';
+	word[8] = 0;
+	p = pin;
+	i = 6;
+	while(p) {
+		word[i] = '0' + (p % 10);
+		p = p / 10;
+		--i;
+	}
+	p = pin;
+	i = 0;
+	while(p) {
+		i = i + (3 * (p % 10));
+		p = p / 10;
+		i = i + p % 10;
+		p = p / 10;
+	}
+	word[7] = '0' + ((10 - i % 10) % 10);
+	++pin;
+}
+void restore() {
+}
```

Comment:
Bug: defining `restore()` tells JtR that this format can restore, whereas it doesn't actually infer the value of `pin` from `word` in that function. I suggest you simply drop `restore()`.